colossalai/schema/zero.py

```diff
@@ -0,0 +1,42 @@
+from pydantic import (BaseModel, StrictBool, StrictInt, StrictFloat, validator)
+from typing import Any
+
+__all__ = ['ZeroConfig']
+
+
+class ZeroModelConfig(BaseModel):
+    # TODO: resolve circular import and replace Any with BaseShardStrategy
+    shard_strategy: Any
+    reduce_scatter_bucket_size_mb: int = 25
+    fp32_reduce_scatter: StrictBool = False
+    offload_config: dict = dict(device="cpu")
+    gradient_predivide_factor: StrictFloat = 1.0
+    use_memory_tracer: StrictBool = False
+    reuse_fp16_shard: StrictBool = False
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    @validator('offload_config')
+    def check_offload_device(cls, v):
```

Comment:
We should leave the check as a member function in the ShardedModel or corresponding class.
Therefore if we change the interface, we do not have to remember to change the logic here.

Reply:
Got it.